data-raw/R2R_TMH_habitat_inputs/cache_tmh_dataR.R

```diff
@@ -0,0 +1,306 @@
+library(tidyverse)
+
+# file to cache Theoretical Max Habitat data objects 
+# and some exploratory plots to compare SIT existing 
+# to TMH
+
+all_habitat_data <- readRDS('data-raw/R2R_TMH_habitat_inputs/all_habitat_data_for_tmh_inputs_fall_run.rdata')
+
+all_hab_data_long <- all_habitat_data |>
+  mutate(watershed = ifelse(watershed == "Lower San Joaquin River", "San Joaquin River", watershed)) |> 
+  rename(spwn_acres_max = max_spawning_acres,
+         rear_acres_max = max_rearing_acres, 
+         flood_acres_max = max_floodplain_acres) |> 
+  pivot_longer(cols = c(spwn_flow:flood_acres_max), names_to = "metric") |>
+  separate(metric, c('hab', 'unit', 'lifestage'), "_") |>
+  mutate(max_hab = ifelse(lifestage == "max", "max_hab", NA),
+         lifestage = ifelse(lifestage == "max", NA, lifestage),
+         unit = ifelse(unit == "flow", "cfs", unit)) |> 
+  glimpse()
+
+
+# update DSMhabitat values ------------------------------------------------
+
+watersheds <- unique(all_hab_data_long$watershed)
+watersheds <- watersheds[!(watersheds %in%  c('North Delta', "South Delta"))]
+
+## spawning ----------------------------------------------------------------
+
+r_to_r_tmh_fr_spawn <- DSMhabitat::fr_spawn$biop_itp_2018_2019
+
+for(i in 1:length(watersheds)) {
```

Comment:
Let's functionalize these and map through watersheds

Reply:
Then we do not have to repeat for each life stage we can just have that be an argument

Reply:
I will add this to my TODO but if merging is a priority I think functionalizing this part can happen later if that's okay with you.